server/graphql/schema/ingestion/resolvers/mutations/ingestData.ts

```diff
@@ -288,20 +288,57 @@ class IngestDataWorker extends ResolverBase {
     }
 
     private async validateIdentifier(identifier: IngestIdentifierInput): Promise<IdentifierResults> {
-        // compute identifier; for ARKs, extract the ID from a URL that may be housing the ARK ID
+
+        const regexArk : RegExp = new RegExp("ark:\/65665\/[0-9a-zA-Z-]*([^@!%$&#*+=.,\"\'\^]+$)");
+        const regexEdan : RegExp = new RegExp("edanmdm\:[a-z]*[a-z|A-Z|0-9_.]*([^@!%$&#*=+\"\']+$)");
+
+        // ARK
         const vocabularyARK: DBAPI.Vocabulary | undefined = await this.getVocabularyARK();
         if (!vocabularyARK)
             return { success: false, error: 'Unable to compute ARK Vocabulary ID' };
-        let identifierValue: string;
+
+        let identifierValue: string = '';
+
         if (identifier.identifierType == vocabularyARK.idVocabulary) {
             const arkId: string | null = this.getICollection().extractArkFromUrl(identifier.identifier);
-            if (!arkId)
-                return { success: false, error: `Invalid ark ${identifier.identifier}` };
+            
+            // Uncomment to Console Test in Terminal
+            // console.log({ regexTestValue: regexArk.test(arkId ?? ""), arkValue: arkId })
+
+            if  (!arkId)
+                return { success: false, error: `Invalid Ark ID: ${identifier.identifier}` };
+            
+            if (regexArk.test(arkId))
+                identifierValue = arkId;   
             else
-                identifierValue = arkId;
-        } else
+                return {success: false, error: `Incorrect Ark Record ID Syntax.  Please review documentation.`}   
+        } 
+
+        //  EDAN
+        const vocabularyEdan: DBAPI.Vocabulary | undefined = await this.getVocabularyEdanRecordID();
+        if (!vocabularyEdan) 
+            return {success: false, error: `Unable to compute EDAN Vocabulary ID.`}
+        
+        if (identifier.identifierType == vocabularyEdan.idVocabulary) {
+            const edanId: string | null = this.getICollection().checkEdanIdentifier(identifier.identifier); 
+
+            //Uncomment to Console Test in Terminal
+            //console.log({ regexTestValue: regexEdan.test(edanId ?? ""), edanValue: edanId })
+
+            if  (!edanId)
+                return { success: false, error: `Invalid EDAN Record ID: ${identifier.identifier}}` };
+             
+            if (regexEdan.test(edanId))
```

Comment:
I'm noticing the newly added ICollection.checkEdanIdentifier().  Why isn't that being used here, instead of regexEdan?